blocks/query-heading/edit.tsx

```diff
@@ -1,17 +1,119 @@
-import { __ } from '@wordpress/i18n';
-import { useBlockProps } from '@wordpress/block-editor';
+import apiFetch from '@wordpress/api-fetch';
+import { BlockControls, RichText, useBlockProps } from '@wordpress/block-editor';
+import { Spinner, ToolbarDropdownMenu, ToolbarGroup } from '@wordpress/components';
+import { useEffect, useState } from '@wordpress/element';
+import {
+  heading,
+  headingLevel1,
+  headingLevel2,
+  headingLevel3,
+  headingLevel4,
+  headingLevel5,
+  headingLevel6,
+} from '@wordpress/icons';
+import { __, sprintf } from '@wordpress/i18n';
+import { addQueryArgs } from '@wordpress/url';
 
-import './index.scss';
+interface EditProps {
+  attributes: {
+    level?: number,
+    override?: string;
+  };
+  context?: {
+    curation?: any; // Shape TBD.
```

Comment:
did you want to flesh this out yet?

Reply:
I'm not sure what the shape will be...I think I'm confused about how this context works.